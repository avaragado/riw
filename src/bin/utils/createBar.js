// @flow

import chalk from 'chalk';

const sBlock = '⣿';
const arsPart = ['', '⡀', '⡄', '⡆', '⡇', '⣇', '⣧', '⣷'];

export default (ctTotal: number, ctChar: number = 20) => {
    const ctBlockPerNum = ctChar / ctTotal;
    const ctPipPerNum = (ctChar * arsPart.length) / ctTotal;

    const bar = num => sBlock.repeat(num * ctBlockPerNum);
    const part = num => arsPart[Math.floor(num * ctPipPerNum) % arsPart.length];
    const barspace = num => (bar(num) + part(num)).padEnd(ctChar);
    const slash = num => `${num}/${ctTotal}`;

    return (num: number) => `${chalk.green(barspace(num))} ${chalk.dim(slash(num))}`;
};
