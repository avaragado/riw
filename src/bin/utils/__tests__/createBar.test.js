// @flow

import chalk from 'chalk';

import createBar from '../createBar';

type Fixture = {
    name: string,
    in: {
        ctTotal: number,
        ctChar: number,
        num: number,
    },
};

const fixtures: Fixture[] = [
    {
        name: '01',
        in: {
            ctTotal: 100,
            ctChar: 10,
            num: 0,
        },
    },

    {
        name: '02',
        in: {
            ctTotal: 200,
            ctChar: 20,
            num: 200,
        },
    },

    {
        name: '03',
        in: {
            ctTotal: 100,
            ctChar: 10,
            num: 50,
        },
    },

    {
        name: '04',
        in: {
            ctTotal: 80,
            ctChar: 10,
            num: 19,
        },
    },

    {
        name: '05',
        in: {
            ctTotal: 80,
            ctChar: 10,
            num: 20,
        },
    },

    {
        name: '06',
        in: {
            ctTotal: 100,
            ctChar: 10,
            num: 1,
        },
    },

    {
        name: '07',
        in: {
            ctTotal: 100,
            ctChar: 10,
            num: 2,
        },
    },

    {
        name: '08',
        in: {
            ctTotal: 100,
            ctChar: 10,
            num: 54,
        },
    },
];

describe('bin/utils/createBar', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            const received = createBar(fixture.in.ctTotal, fixture.in.ctChar)(fixture.in.num);

            expect(received).toMatchSnapshot();
        });
    });
});
