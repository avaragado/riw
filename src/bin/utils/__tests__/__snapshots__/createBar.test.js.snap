// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bin/utils/createBar 01 1`] = `"[32m          [39m [2m0/100[22m"`;

exports[`bin/utils/createBar 02 1`] = `"[32mâ£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿â£¿[39m [2m200/200[22m"`;

exports[`bin/utils/createBar 03 1`] = `"[32mâ£¿â£¿â£¿â£¿â£¿     [39m [2m50/100[22m"`;

exports[`bin/utils/createBar 04 1`] = `"[32mâ£¿â£¿â¡†       [39m [2m19/80[22m"`;

exports[`bin/utils/createBar 05 1`] = `"[32mâ£¿â£¿â¡‡       [39m [2m20/80[22m"`;

exports[`bin/utils/createBar 06 1`] = `"[32m          [39m [2m1/100[22m"`;

exports[`bin/utils/createBar 07 1`] = `"[32mâ¡€         [39m [2m2/100[22m"`;

exports[`bin/utils/createBar 08 1`] = `"[32mâ£¿â£¿â£¿â£¿â£¿â¡†    [39m [2m54/100[22m"`;
